gemini/sample-apps/accelerating_product_innovation/app/pages_utils/utils_edit_image.py

```diff
@@ -0,0 +1,275 @@
+    background_image: Image.Image,
+    bg_editing: bool = False,
+) -> bytes:
+    """
+    Processes a foreground image, optionally removing white regions,
+    and prepares it for merging with a background image.
+
+    Args:
+        foreground_image (Image.Image): The PIL Image object representing the
+        foreground.
+        background_image (Image.Image): The PIL Image object representing the
+        background.
+        bg_editing (bool, optional): If True, removes white regions from the
+        foreground. Defaults to False.
+
+    Returns:
+        bytes: The processed and merged image data as bytes.
+    """
+
+    # Logic to edit background (invert mask)
+    if bg_editing:
+        # Get image foreground
+        foreground_data = foreground_image.getdata()
+        new_bytes = []
+        # Invert pixels.
+        for item in foreground_data:
+            if item[0] == 255 and item[1] == 255 and item[2] == 255:
+                new_bytes.append((255, 255, 255, 0))
+            else:
+                new_bytes.append((255, 255, 255, 1))
+
+        foreground_image.putdata(new_bytes)
+
+    # Resize and merge foreground with background
+    resized_foreground = foreground_image.resize(background_image.size)
+    merged_image = background_image.copy()
+    merged_image.paste(resized_foreground, (0, 0), resized_foreground)
+
+    # Convert to bytes for storage
+    with io.BytesIO() as buffer:
+        merged_image.save(buffer, format="PNG")
+        processed_image_bytes = buffer.getvalue()
+
+    return processed_image_bytes
+
+
+def initialize_edit_page_state() -> None:
+    """
+    Initializes the session state for the image editing page.
+
+    This function checks if the session state has been initialized, and if not,
+    it initializes it.
+    It also checks if an image has been uploaded, and if so, it sets
+    the session state accordingly.
+    """
+
+    # Check which image file prefix points to the image to be edited
+    if "image_to_edit" not in st.session_state or st.session_state.image_to_edit == -1:
+        st.session_state.image_to_edit = (
+            -1
+        )  # No image from generations is being edited.
+        st.session_state.image_file_prefix = (
+            "uploaded_image"  # image prefix for editing uploaded image.
+        )
+        st.session_state.uploaded_img = True  # Set image uploaded to true.
+    else:
+        st.session_state.uploaded_img = False  # Generated image being edited.
+        st.session_state.start_editing = True  # Display canvas for editing.
+
+
+def handle_image_upload() -> None:
+    """
+    Handles an image upload, saving the image and updating session state.
+    """
+    # Uplad button
+    uploaded_file = st.file_uploader("Upload an image")
+
+    # If image has been uploaded open and Save uploaded image.
+    if uploaded_file is not None:
+        try:
+            image = Image.open(uploaded_file)
+            filename = "uploaded_image0.png"
```

Comment:
We are keeping this as a temporary file, which is needed only while editing. In case a new image is uploaded, existing file is overwritten, as it won't be needed.